src/components/pill/pill.pw.tsx

```diff
@@ -0,0 +1,427 @@
+    });
+  });
+
+  ([
+    ["tag", tag],
+    ["status", status],
+  ] as [PillProps["pillRole"], string][]).forEach(([role, output]) => {
+    test(`should render pillRole prop set as ${role}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<PillComponent pillRole={role}>{role}</PillComponent>);
+
+      await checkCSSOutline(
+        pillPreview(page),
+        "2px",
+        "border",
+        "solid",
+        output
+      );
+    });
+  });
+
+  ([
+    [colorsSemanticCaution500, "rgb(239, 103, 0)"],
+    [blackOpacity65, "rgba(0, 0, 0, 0.65)"],
+    [brilliantGreenShade20, "rgb(0, 176, 0)"],
+    [red, "rgb(255, 0, 0)"],
+    [hexBlue, "rgb(18, 52, 86)"],
+    [green, "rgb(0, 123, 10)"],
+  ] as const).forEach(([color, output]) => {
+    test(`should render borderColor prop set as ${output}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(
+        <PillComponent pillRole="status" borderColor={color}>
+          Pill
+        </PillComponent>
+      );
+
+      await checkCSSOutline(
+        pillPreview(page),
+        "2px",
+        "border",
+        "solid",
+        output
+      );
+    });
+  });
+
+  ["20px", "100px"].forEach((maxWidth) => {
+    test(`should render maxWidth prop set to ${maxWidth}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<Pill maxWidth={maxWidth}>Pill with a long label</Pill>);
+
+      const elementLocator = pillPreview(page);
+      await expect(elementLocator).toHaveCSS("max-width", maxWidth);
+    });
+  });
+
+  ([
+    [small, "16px", "12px", "0px 8px"],
+    [medium, "20px", "14px", "0px 8px"],
+    [large, "24px", "14px", "0px 8px"],
+    [extraLarge, "28px", "16px", "0px 12px"],
+  ] as const).forEach(([size, height, fontSize, padding]) => {
+    test(`should render size prop set as ${size}`, async ({ mount, page }) => {
+      await mount(<Pill size={size}>Pill</Pill>);
+
+      const elementLocator = pillPreview(page);
+
+      await expect(elementLocator).toHaveCSS("min-height", height);
+
+      await expect(elementLocator).toHaveCSS("line-height", height);
+
+      await expect(elementLocator).toHaveCSS("font-size", fontSize);
+
+      await expect(elementLocator).toHaveCSS("padding", padding);
+    });
+  });
+
+  ([
+    [true, "break-spaces"],
+    [false, "nowrap"],
+  ] as const).forEach(([booleanState, cssValue]) => {
+    test(`should render wrapText prop set as ${booleanState}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(
+        <Pill maxWidth="44px" wrapText={booleanState}>
+          Wrapped pill
+        </Pill>
+      );
+
+      const elementLocator = pillPreview(page);
+      await expect(elementLocator).toHaveText("Wrapped pill");
+      await expect(elementLocator).toHaveCSS("white-space", cssValue);
+    });
+  });
+});
+
+test.describe("should check focus outlines and border radius", () => {
+  (["S", "M", "L", "XL"] as PillProps["size"][]).forEach((size) => {
```

Comment:
As these sizes already have constants declared as `small`, `medium`, `large` and `extraLarge`, could you not just use them here rather than stating the values?